addons/test_discuss_full/tests/test_performance.py

```diff
@@ -304,7 +305,7 @@ def test_init_messaging(self):
                             'display_name': 'Ernest Employee',
                             'email': 'e.e@example.com',
                             'id': self.users[0].partner_id.id,
-                            'im_status': 'offline',
+                            'im_status': 'leave_offline' if self.users[0].employee_id.is_absent else 'offline',
```

Comment:
Isn't it possible to determine in advance according to the test setup whether the result will be `leave_offline` or `offline`?

Reply:
@seb-odoo That's what I suspected, however there must be a reason that `out_of_office_date_end` is computed on the fly? I was hoping you could enlighten me :slightly_smiling_face:

Reply:
I believe `out_of_office_date_end` is computed to not bother with mock date in this test.
But using a frozen date and hardcoded value would probably be easier.

Reply:
Well, the end date is available in the setup is it not? The leave is created with `'date_to': date.today() + relativedelta(days=2),`. Am I missing something or would it be possible to simply use this value?

Reply:
Using that kind of value without enforcing a frozen date could lead to issues on runbot (especially around midnight).